src/utils/functions.ts

```diff
@@ -284,6 +287,58 @@ const generateShareTemplate = (
   return baseMessage;
 };
 
+// fetches playlist data (metadata and videos)
+const fetchPlaylistData = async (
+  playlistId: string,
+  pageToken: string = '',
+  accumulatedVideos: Video[] = [],
+  metadata: { playlistName?: string; description?: string } = {}
+): Promise<PlaylistModel> => {
+  const response = await fetch(
+    `${YOUTUBE_API_PATH}/playlistItems?part=snippet&playlistId=${playlistId}&maxResults=50&pageToken=${pageToken}&key=${process.env.YOUTUBE_API_KEY}`
+  );
+  const data = await response.json();
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch playlist data: ${data.error.message}`);
+  }
+
+  // Extract playlist metadata if not already set
+  if (!metadata.playlistName && data.items.length > 0) {
+    metadata.playlistName = data.items[0].snippet.channelTitle || 'Unknown Playlist Name';
+    metadata.description = data.items[0].snippet.description || 'No Description Available';
+  }
+
+  // Extract video details
+  const videos: Video[] = data.items.map((item: any) => ({
+    title: item.snippet.title,
+    videoId: item.snippet.resourceId.videoId,
+    thumbnail: item.snippet.thumbnails?.default?.url || 'https://via.placeholder.com/150',
+  }));
+
+  // Accumulate videos
+  const allVideos = [...accumulatedVideos, ...videos];
+
+  // Continue fetching if there's a nextPageToken
+  if (data.nextPageToken) {
+    return fetchPlaylistData(playlistId, data.nextPageToken, allVideos, metadata);
+  }
+
+  // Return the complete data when no more pages
+  return {
+    playlistName: metadata.playlistName || 'Unknown Playlist Name',
+    description: metadata.description || 'No Description Available',
+    videos: allVideos,
+  };
+};
```

Comment:
No need to use Deafult Playlist name. If No Playlist name simply return Not found